modules/transactionpool/update.go

```diff
@@ -39,6 +46,46 @@ func (tp *TransactionPool) ProcessConsensusChange(cc modules.ConsensusChange) {
 	if err != nil {
 		tp.log.Println("ERROR: could not update the recent consensus change:", err)
 	}
+	err = tp.putBlockHeight(tp.dbTx, tp.blockHeight)
+	if err != nil {
+		tp.log.Println("ERROR: could not update the block height:", err)
+	}
+
+	// prune transactions older than maxTxnAge.
+	pruneTxns := make(map[types.TransactionID]struct{})
+	for txid, seenHeight := range tp.transactionHeights {
+		if tp.blockHeight-seenHeight > maxTxnAge {
+			delete(tp.transactionHeights, txid)
+			pruneTxns[txid] = struct{}{}
+		}
+	}
+	for id, tSet := range tp.transactionSets {
+		var validTxns []types.Transaction
+		for _, txn := range tSet {
+			_, shouldPrune := pruneTxns[txn.ID()]
+			if !shouldPrune {
```

Comment:
instead of checking `shouldPrune`, can't you just check tp.transactionHeights inline?